scripts/e2e/external/setup.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
```

Comment:
Duplicating this logic here is going to complicate maintenance. If these scripts were in `shipyard/scripts/shared/` they would at least not be in two different repos, and they could be used in CI other than in the submariner-operator repo.

Reply:
@dfarrell07 

Thank you for your feedback.

Below are still just an idea, though:

- For setup scripts, current implentation is just calling two extra bash functions, which can be merged into one, before deploying submariner. So, if shipyard has an extension mechanism to call pre/post deploy functions, these duplicated codes can be removed,
- For e2e scripts, we can use `ginkgo.focus` and `ginkgo.skip` to filter out external network tests. So, we would be able to share the same e2e_test.go, and make e2e scripts focus for normal e2e and e2e-external.
(Then, [this](https://github.com/submariner-io/shipyard/pull/502) might not be needed.)

I'll start working on implementing it, to see if it is actually doable.

Reply:
Yes, that makes perfect sense to me @mkimuram, thank you! :+1: :pray:

Reply:
Could we use the cluster args to indicate to clusters.sh that we're in this external clusters situation, and that those pre/post deploy functions need to be used? Then we could use an interface to the E2E GHA similar to https://github.com/submariner-io/shipyard/pull/487.